components/cdc/src/delegate.rs

```diff
@@ -606,13 +606,16 @@ impl Delegate {
                 }
                 "lock" => {
                     let mut row = EventRow::default();
-                    let skip = decode_lock(put.take_key(), put.get_value(), &mut row);
+                    let lock = Lock::parse(put.get_value()).unwrap();
+                    let for_update_ts = lock.for_update_ts;
+                    let skip = decode_lock(put.take_key(), lock, &mut row);
                     if skip {
                         continue;
                     }
 
                     if self.txn_extra_op == TxnExtraOp::ReadOldValue {
-                        let key = Key::from_raw(&row.key).append_ts(row.start_ts.into());
+                        let key = Key::from_raw(&row.key)
+                            .append_ts(std::cmp::max(for_update_ts, row.start_ts.into()));
```

Comment:
Will the cache be queried with "key+for_update_ts"? But the DeltaScanner caches it with "key+start_ts". Is there a problem?